### ISSUE TITLE
optional `r#type` yields invalid identifier

### ISSUE THREAD

**Issue by oktal** [elastio/bon#174]
Hello,

I think I hit an edge-case that #9 does not properly handle.

The following struct definition:

```rs
enum VariableType {
    Flow,
    Temperature,
    Pressure,
}

#[derive(bon::Builder)]
struct Variable {
    r#type: Option<VariableType>,
    name: String,
    key: String,
}
```

Will panic with the following error:

```
proc-macro derive panicked
   message: `"maybe_r#type"` is not a valid identifier
```

It works if I remove the `Option` or rename my field to `variable_type`

### A note for the community from the maintainers

Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to help the maintainers with prioritizing it. You may add a comment describing your real use case related to this issue for us to better understand the problem domain.

**Comment by Veetaha** [comment]
Thank you for reporting this!

The fix is pending a 3.0 release (https://github.com/elastio/bon/issues/156)

**Comment by Veetaha** [comment]
`3.0.0-rc` version was published. I'll prepare a blog post for the release and switch it to `3.0.0` on November 13-th
